src/ast.rs

```diff
@@ -12,6 +12,27 @@ pub use expr::*;
 
 use halo2_proofs::plonk::{Advice, Column as Halo2Column, ColumnType, Fixed};
 
+use self::query::Queriable;
+
+/// Signal abstraction
+#[derive(Clone, Debug)]
+pub enum Signal {
```

Comment:
I don't think we should have a new enum just for the exposed feature, just check in ast::expose could recieve a Queriable, and then panic for when no forward or shared.

Fixed signal cannot be exposed.